FindANSAK.cmake

```diff
@@ -105,317 +108,357 @@ For DevStudio Generators, the picture is slightly murkier. See following comment
 
 ############################################################################
 # announce settings if there were any
-if(ANSAK_DIR)
-    message(STATUS "var ANSAK_DIR is defined: ${ANSAK_DIR}")
+if( ANSAK_DIR )
+    message( STATUS "var ANSAK_DIR is defined: ${ANSAK_DIR}" )
 endif()
 
-if(NOT "$ENV{ANSAK_DIR}" STREQUAL "")
-    message(STATUS "environment ANSAK_DIR is defined: $ENV{ANSAK_DIR}")
+if( NOT "$ENV{ANSAK_DIR}" STREQUAL "" )
+    message( STATUS "environment ANSAK_DIR is defined: $ENV{ANSAK_DIR}" )
 endif()
 
-if(PROGRAM_DATA)
-    message(STATUS "var PROGRAM_DATA is defined: ${PROGRAM_DATA}")
+if( PROGRAM_DATA )
+    message( STATUS "var PROGRAM_DATA is defined: ${PROGRAM_DATA}" )
 endif()
 
-if(NOT "$ENV{PROGRAM_DATA}" STREQUAL "")
-    message(STATUS "environment PROGRAM_DATA is defined: $ENV{PROGRAM_DATA}")
+if( NOT "$ENV{PROGRAM_DATA}" STREQUAL "" )
+    message( STATUS "environment PROGRAM_DATA is defined: $ENV{PROGRAM_DATA}" )
 endif()
 
-if(NOT "$ENV{ANSAK_NO_GENERATOR_LIBS}" STREQUAL "" OR ANSAK_NO_GENERATOR_LIBS)
-    set(_no_generator_libs True)
+if( NOT "$ENV{ANSAK_NO_GENERATOR_LIBS}" STREQUAL "" OR ANSAK_NO_GENERATOR_LIBS )
+    set( _no_generator_libs True )
 else()
-    set(_no_generator_libs False)
+    set( _no_generator_libs False )
 endif()
 
-if (ANSAK_DEBUG)
-    message("ANSAK_DIR=\"$ENV{ANSAK_DIR}/${ANSAK_DIR}\"")
-    message("PROGRAM_DATA=\"$ENV{PROGRAM_DATA}/${PROGRAM_DATA}\"")
-    message("ANSAK_NO_GENERATOR_LIBS=$ENV{ANSAK_NO_GENERATOR_LIBS}/${ANSAK_NO_GENERATOR_LIBS}")
+if ( ANSAK_DEBUG )
+    message( "ANSAK_DIR=\"$ENV{ANSAK_DIR}/${ANSAK_DIR}\"" )
+    message( "PROGRAM_DATA=\"$ENV{PROGRAM_DATA}/${PROGRAM_DATA}\"" )
+    message( "ANSAK_NO_GENERATOR_LIBS=$ENV{ANSAK_NO_GENERATOR_LIBS}/${ANSAK_NO_GENERATOR_LIBS}" )
 endif()
 #
 ############################################################################
 
-set(win32Dirs Win32 x86 i386 X86)
-set(win64Dirs x64 Win64 x86-64 x86_64 X64 X86_64 X86-64 amd64)
-set(armDirs ARM Arm arm)
-set(arm64Dirs ARM64 Arm64 arm64)
-function(platform_matches_dirname dirname result)
-    if(CMAKE_GENERATOR_PLATFORM STREQUAL "ARM")
-        set(theList ${armDirs})
-    elseif(CMAKE_GENERATOR_PLATFORM STREQUAL "ARM64")
-        set(theList ${arm64Dirs})
-    elseif(CMAKE_GENERATOR_PLATFORM STREQUAL "Win32")
-        set(theList ${win32Dirs})
-    elseif(CMAKE_GENERATOR_PLATFORM STREQUAL "x64")
-        set(theList ${win64Dirs})
+set( win32Dirs Win32 x86 i386 X86 )
+set( win64Dirs x64 Win64 x86-64 x86_64 X64 X86_64 X86-64 amd64 )
+set( armDirs ARM Arm arm )
+set( arm64Dirs ARM64 Arm64 arm64 )
+function( platform_matches_dirname dirname result )
```

Comment:
Function to help find `C:\ProgramData\lib\<platform>\<config>` as a place where libraries can be found by MSDev-generator-written solution files.